### PR TITLE
Training request review process field

### LINKED ISSUE #1501: Instructor Training Application checkbox to opt-in for Open review process
This is not urgent, but if it is a quick fix, it would be a fantastic improvement.

Currently, applications are routed to our Open Instructor Training program when they are missing a Registration Code. This means that anyone who forgets to enter a code ends up taking time for unnecessary application review.

It would be much more straightforward to have a checkbox with text like this: "OPEN INSTRUCTOR TRAINING APPLICANTS: Check here if you would like this application to be reviewed for admission to our Open Instructor Training program. If you have already been invited to attend a training, please do not select this box."

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
This fixes #1501 by adding a new form field: review process for selecting whether user follows pre-approved registration process, or individual training application process.

This PR very possibly needs some language tuning, however the needed mechanics (like HTML widgets on the form and proper validation) are all here.

I based this PR on the newest 3 forms project. That's why there's so many commits here. But don't worry, they will be gone once we merge that PR (#1525).

Below I present some screenshots and highlight the wording I'm concerned about:

obraz

obraz

1. Yellow: field's label
2. Blue: field's options - I based them on https://github.com/carpentries/amy/issues/1501#issuecomment-518298094
3. Pink: notes for each option - I based them on https://github.com/carpentries/amy/issues/1501#issuecomment-518298094
4. Orange: I cut some text from that help text (original: "If you are scheduled to receive training at a member site, please enter the same code you were given to register on Eventbrite. Otherwise please leave this blank.")

I'm open for suggestions on verbiage for each of these.

--------------

Last thing I'm concerned about is a prefilling registration code feature. If you go to https://amy.segmars.com/forms/request_training/ you'll find empty "Registration code" field. However, if you go to https://amy.carpentries.org/forms/request_training/?group=gummybears then the field is gone (it's hidden).

I extended this behavior - now providing "?group" in URL preselects "Pre-approved Registration" option, but the "Registration code" is still hidden. It looks like this:
obraz

I need to know if we want to keep hiding the "Registration code" field when "?group" is provided in URLs.

Given sheer feedback I'm waiting for, I don't think this will land in v2.9, especially since we're releasing it on Tuesday 17.09.2019.

----------

To sum up:
1. I need feedback on wording (field's label, options and their notes, and one help text for "Registration code" field)
2. I need feedback on prefilling group name / registration code - do we keep it? Do we keep hiding the field?
3. I don't think I can manage to merge this into v2.9, because it's very soon to be released.

----------

EDIT: Added some horizontal lines for better reading.

### CODE REVIEW

**Comment by maneesha** [comment]
Orange: If you are scheduled to receive training at a member site, please enter the same code you were given to register on Eventbrite.  > If you have been given a registration code through a Carpentries member site or for a specific scheduled event, please enter it here:  
(change in wording needed as not all events register via Eventbrite) 

Pink: "If your organisation has Membership..."  > "If you have been invited to apply through an institutional membership or other agreement with The Carpentries" 

Yellow: "Review process workflow"  -- I think this wording should be changed, but I need some time to think about what this should be.  Suggestions from @karenword  are welcome!

Blue: "Individual Open Training Application" > "Open Training Application"

**Comment by maneesha** [comment]
We have never used the feature to include a url parameter for filling in a group code, so I think it's fine to deprecate that.

**Comment by karenword** [comment]
Thank you @pbanaszkiewicz ! 
I think that, as-is, this is better than what we currently have and I do not foresee it causing any major problems. Unless @maneesha disagrees, I think it would be best to include this in the current release if possible, and tweak language thereafter as needed.

To the extent that there is time to address feedback:

- I think "Review process workflow" could be something like "Application Type" 
- Otherwise agree with Maneesha's suggestions.

**Comment by pbanaszkiewicz** [comment]
@karenword @maneesha  since this PR is based on 3 forms project, I decided to host this branch on the server, so that you can check it out yourselves.

https://amy.segmars.com/forms/request_training/
